### ISSUE TITLE
🐛Bug - Rules missing

### ISSUE THREAD

**Issue by pierssinclairssw** [SSWConsulting/SSW.Rules.Widget#36]
Hi @BrookJeynes @bradystroud @Aibono1225 

The rules widget is broken because it's missing rules.

![image](https://github.com/SSWConsulting/SSW.Rules.Widget/assets/79821522/642a9431-7836-4931-bb93-6008261039ed)
**Figure: I created this rule 2 months ago**

![image](https://github.com/SSWConsulting/SSW.Rules.Widget/assets/79821522/4790094e-c441-49e0-bf87-86632b719493)
**Figure: Rule is not there**

Acceptance Criteria:
The rules widget should display all rules edited or created by a user

**Comment by pierssinclairssw** [comment]
Hey @Aibono1225 I just wrote a new rule and it seems to be suffering from the same problem that it doesn't show up in the widget:
https://www.ssw.com.au/rules/relational-database-design/

**Comment by Aibono1225** [comment]
Update after investigation, here is what @bradystroud and I found for this issue:

Context: We refactored the code for the widget so that it requests data from the widget-history.json instead of making multiple requests to the GitHub. See the PR: https://github.com/SSWConsulting/SSW.Rules.Widget/pull/31

We found that there are problems with the data source, the data in the widget-history.json is not complete.
![image](https://github.com/SSWConsulting/SSW.Rules.Widget/assets/127192800/7833dc3f-73a8-40eb-989d-676248084b37)
**Figure:  Why is the Widget missing some rules**

There are two possible reasons for incomplete data in the database:
1. When someone modifies an existing rule, the previous information about the person making the changes will be replaced (Database update).  https://github.com/SSWConsulting/SSW.Rules.Functions/blob/d9a51836bcf17bf9ae3fce1503c54200ff466bbf/Functions/Functions/History/UpdateRuleHistory.cs#L63

2. If "UpdateRuleHistory" fails, the execution of "UpdateHistorySyncCommitHash" will still continue, hence any missed data will not be readded to the CosmosDB.
https://github.com/SSWConsulting/SSW.Rules/blob/34f46dd9b2551c1fba89524146e2764c21b62ddd/.azure/scripts/update-rule-history.ps1#L69C1-L83
